internal/services/azapi_data_plane_resource.go

```diff
@@ -0,0 +1,242 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/terraform-provider-azapi/internal/clients"
+	"github.com/Azure/terraform-provider-azapi/internal/locks"
+	"github.com/Azure/terraform-provider-azapi/internal/services/parse"
+	"github.com/Azure/terraform-provider-azapi/internal/services/validate"
+	"github.com/Azure/terraform-provider-azapi/internal/tf"
+	"github.com/Azure/terraform-provider-azapi/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceAzApiDataPlaneResource() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAzApiDataPlaneResourceCreateUpdate,
+		Read:   resourceAzApiDataPlaneResourceRead,
+		Update: resourceAzApiDataPlaneResourceCreateUpdate,
+		Delete: resourceAzApiDataPlaneResourceDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				return nil, fmt.Errorf("importing is not supported for this resource yet")
```

Comment:
Do we have a specific reason to not support importing?

Reply:
It's complicated to pull the data plane resource type from the ID, I'll implement this feature in another PR/release.